src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -515,22 +519,27 @@ static size_t uct_cuda_copy_md_get_total_device_mem(CUdevice cuda_device)
 static void
 uct_cuda_copy_sync_memops(uct_cuda_copy_md_t *md, const void *address)
 {
+    unsigned value = 1;
+
 #if HAVE_CUDA_FABRIC
     ucs_status_t status;
-    if (!md->sync_memops_set) {
-        /* Synchronize future DMA operations for all memory types */
-        status = UCT_CUDADRV_FUNC_LOG_WARN(cuCtxSetFlags(CU_CTX_SYNC_MEMOPS));
-        if (status == UCS_OK) {
-            md->sync_memops_set = 1;
+    if (md->config.cuda_ctx_set_flags) {
+        if (!md->sync_memops_set) {
+            /* Synchronize future DMA operations for all memory types */
+            status = UCT_CUDADRV_FUNC_LOG_WARN(cuCtxSetFlags(CU_CTX_SYNC_MEMOPS));
```

Comment:
I think we should use here `cuGetProcAddress`. So we can check the presence of the function in runtime.

Reply:
agree seems better (we might still need to disable fabric?)

Reply:
Yes, we should.
We need to check the availability of the method. If it is not there, then disable fabric.

If we leave a direct call to `cuCtxSetFlags`, we will still have issues in runtime.
That is why we have to use `cuGetProcAddress` to retrieve pointer to the function. And use it to set flags.

Reply:
fixed